src/tag.ts

```diff
@@ -106,19 +116,22 @@ export class Tag implements IStateful<ITagState> {
     }
   }
 
-  public id: string|undefined
-  public coreState: ITagState|undefined
-
-  constructor(
-    public context: Arc,
-    idOrOpts: string|ITagState
-  ) {
-    if (typeof idOrOpts === 'string') {
-      this.id = idOrOpts
-    } else {
-      this.id = idOrOpts.id
-      this.setState(idOrOpts)
+  public static itemMap = (context: Arc, item: any): Tag => {
+    if (item === null) {
+      //TODO: How to get ID for this error msg?
```

Comment:
hmmm seems to me we need to do some error handling better with the queries. Maybe have an explicity error handler that can send a good error saying what the query was so the developer can get the error and try on their own to see what's wrong.

Reply:
FYI We have an outstanding issue in the client on trying to provide more informative messages, particularly on contract reverts.

Reply:
@namesty I think we should have the query be passed into the itemMap so that in this error case, we can emit the query that was used.

Another option is to pass in the entity, so you can get the ID that way.